src/olympia/amo/indexers.py

```diff
@@ -0,0 +1,147 @@
+                        'string': {'type': 'string', 'index': 'no'}
+                    }
+                }
+            })
+
+    @classmethod
+    def attach_language_specific_analyzers(cls, mapping, field_names):
+        """
+        For each field in field_names, attach language-specific mappings that
+        will use specific analyzers for these fields in every language that we
+        support.
+
+        These mappings are used by the search filtering code if they exist.
+        """
+        doc_name = cls.get_doctype_name()
+
+        for analyzer in SEARCH_ANALYZER_MAP:
+            if (not settings.ES_USE_PLUGINS and
+                    analyzer in SEARCH_ANALYZER_PLUGINS):
+                log.info('While creating mapping, skipping the %s analyzer '
+                         'as usage of plugins is disabled' % analyzer)
+                continue
+
+            for field in field_names:
+                property_name = '%s_%s' % (field, analyzer)
+                mapping[doc_name]['properties'][property_name] = {
+                    'type': 'string',
+                    'analyzer': analyzer,
+                }
+
+    @classmethod
+    def extract_field_raw_translations(cls, obj, field, db_field=None):
+        """
+        Returns a dict containing raw translations that we need to store for
+        the API.
+        """
+        if db_field is None:
+            db_field = '%s_id' % field
+
+        extend_with_me = {
+            '%s_translations' % field: [
+                {'lang': to_language(lang), 'string': string}
+                for lang, string in obj.translations[getattr(obj, db_field)]
+                if string
+            ]
+        }
+        return extend_with_me
+
+    @classmethod
+    def extract_field_search_translations(cls, obj, field, db_field=None):
+        """
+        Returns a dict containing all translations for the field, that will be
+        analyzed and indexed by ES without language-specific analyzers.
+        """
+        if db_field is None:
+            db_field = '%s_id' % field
+
+        extend_with_me = {
+            field: list(
+                set(s for _, s in obj.translations[getattr(obj, db_field)]))
+        }
+        return extend_with_me
+
+    @classmethod
+    def extract_field_analyzed_translations(cls, obj, field, db_field=None):
+        """
+        Returns a dict containing translations for each language-specific
+        analyzer for the given field.
+        """
+        if db_field is None:
+            db_field = '%s_id' % field
+
+        extend_with_me = {}
+
+        # Indices for each language. languages is a list of locales we want to
+        # index with analyzer if the string's locale matches.
+        for analyzer, languages in SEARCH_ANALYZER_MAP.iteritems():
+            if (not settings.ES_USE_PLUGINS and
+                    analyzer in SEARCH_ANALYZER_PLUGINS):
+                continue
```

Comment:
you have the same check above and you're logging there, can we centralize this check somehow?